src/Http/Http/src/Features/FormFeature.cs

```diff
@@ -177,11 +175,11 @@ private async Task<IFormCollection> InnerReadFormAsync(CancellationToken cancell
         // Some of these code paths use StreamReader which does not support cancellation tokens.
         using (cancellationToken.Register((state) => ((HttpContext)state!).Abort(), _request.HttpContext))
         {
-            var contentType = ContentType;
+            var contentType = _request.ContentType;
             // Check the content-type
-            if (HasApplicationFormContentType(contentType))
+            if (HttpExtensions.HasApplicationFormContentType(contentType))
             {
-                var encoding = FilterEncoding(contentType.Encoding);
+                var encoding = FilterEncoding(ContentType!.Encoding);
```

Comment:
nit: I don't like that the ContentType property allocates a `MediaTypeHeaderValue`. IMO it should be a method.